testsuite/features/step_definitions/command_steps.rb

```diff
@@ -1201,6 +1201,38 @@
   end
 end
 
+Then(/^"([^"]*)" virtual machine on "([^"]*)" should boot using autoyast$/) do |vm, host|
+  node = get_target(host)
+  output, _code = node.run("virsh dumpxml #{vm}")
+  tree = Nokogiri::XML(output)
+  has_kernel = tree.xpath("//os/kernel").size == 1
+  has_initrd = tree.xpath("//os/initrd").size == 1
+  has_autoyast = tree.xpath("//os/cmdline")[0].to_s.include? ' autoyast='
+  raise 'Wrong kernel/initrd/cmdline configuration' unless has_kernel && \
+                                                           has_initrd && has_autoyast
+end
+
+Then(/^"([^"]*)" virtual machine on "([^"]*)" should boot on hard disk at next start$/) do |vm, host|
+  node = get_target(host)
+  output, _code = node.run("virsh dumpxml --inactive #{vm}")
+  tree = Nokogiri::XML(output)
+  has_kernel = tree.xpath("//os/kernel").size == 1
+  has_initrd = tree.xpath("//os/initrd").size == 1
+  has_cmdline = tree.xpath("//os/cmdline").size == 1
+  raise 'Virtual machine will not boot on hard disk at next start' unless !has_kernel && \
+                                                                          !has_initrd && !has_cmdline
+end
+
+Then(/^"([^"]*)" virtual machine on "([^"]*)" should (not stop|stop) on reboot((?: at next start)?)$/) do |vm, host, stop, next_start|
+  node = get_target(host)
+  inactive = next_start == " at next start" ? "--inactive" : ""
```

Comment:
Use single quotes if there is nothing to escape or a variable involved.

Reply:
fixed